src/main/java/net/dv8tion/jda/internal/entities/PrivateChannelImpl.java

```diff
@@ -153,6 +155,17 @@ public MessageAction sendFile(@Nonnull InputStream data, @Nonnull String fileNam
         return PrivateChannel.super.sendFile(data, fileName, options);
     }
 
+    @Nonnull
+    @Override
+    public MessageAction sendFile(@Nonnull File file, @Nonnull String fileName, @Nonnull AttachmentOption... options)
+    {
+        checkBot();
+        final long maxSize = getJDA().getSelfUser().getAllowedFileSize();
+        Checks.check(file.length() <= maxSize,
+            "File may not exceed the maximum file length of %d bytes!", maxSize);
```

Comment:
You have to null check the file here first.

Reply:
Both null checks are called outside of those 2 classes, in MessageChannel, aren't they?

Reply:
You call the supermethod later though, you access the file here before it got null-checked.